### ISSUE TITLE
Confusion with the joints

### ISSUE THREAD

**Issue by andypinxinliu** [nv-tlabs/stmc#4]
I am confused with the smpl joints you have got and its difference with the smplh used in humanml3d. It seems like if you need to call TMR for evaluation, you did

```
x, y, z = T(motions)
motions = T(np.stack((x, z, -y), axis=0))
```

to convert the raw smpl joints in your smplh format to guo's format. May I know the reason why you need to have y to be negative? In the original guo's raw pose preprocessing, after getting the joints, his transform is
```
trans_matrix = np.array([[1.0, 0.0, 0.0],
                            [0.0, 0.0, 1.0],
                            [0.0, 1.0, 0.0]])
```
to let the joints multiply with this matrix, to make y and z axies to be switched. However, when you get the joints, not only did you switch two axies, but also make y to be negative, why is that?

**Comment by Mathux** [comment]
Hello @andypinxinliu, 

Thanks for opening this issue. I think this can help a lot of people working with HumanML3D.

I need y to be negative so that the transformation is a correct rotation. This rotation can be written as:
```
[ 1,  0,  0]
[ 0,  0,  1]
[ 0, -1,  0]
```
which has a determinant of 1. It corresponds to a 90 degree rotation (clockwise) around the x-axis.

In the original guo's raw pose processing, he is using this transform:
```
[ 1,  0,  0]
[ 0,  0,  1]
[ 0,  1,  0]
```
which has a determinant of -1, therefore it is not a valid rotation (but a reflection). But afterward in the guo's pose preprocessing [notebook](https://github.com/EricGuo5513/HumanML3D/blob/main/raw_pose_processing.ipynb), you can see this cell:
```
for i in tqdm(range(total_amount)):
    source_path = index_file.loc[i]['source_path']
    new_name = index_file.loc[i]['new_name']
    data = np.load(source_path)
    start_frame = index_file.loc[i]['start_frame']
    end_frame = index_file.loc[i]['end_frame']
    if 'humanact12' not in source_path:
        if 'Eyes_Japan_Dataset' in source_path:
            data = data[3*fps:]
        if 'MPI_HDM05' in source_path:
            data = data[3*fps:]
        if 'TotalCapture' in source_path:
            data = data[1*fps:]
        if 'MPI_Limits' in source_path:
            data = data[1*fps:]
        if 'Transitions_mocap' in source_path:
            data = data[int(0.5*fps):]
        data = data[start_frame:end_frame]
        data[..., 0] *= -1
    
    data_m = swap_left_right(data)
#     save_path = pjoin(save_dir, )
    np.save(pjoin(save_dir, new_name), data)
    np.save(pjoin(save_dir, 'M'+new_name), data_m)
```
and specifically this line:
```
        data[..., 0] *= -1
```
which does (x -> -x). Those two transformations combined, it corresponds to doing this transformation:
```
[-1,  0,  0]
[ 0,  0,  1]
[ 0,  1,  0]
```
which is now a valid rotation. 

If you compute the difference of the process between mine and guo, it only corresponds to a y-axis rotation of 180 degree. So only the heading direction is different between the two transforms. 
But, after canonicalization in the guo feats code, each frame face one direction (Z+) so it does ends up being exactly the same thing. 

So to recap: I am not using this `trans_matrix` in my code, and prefer using the full rotation directly. This avoid having issues where the left and right arms are swapped (like in the `humanact12` subset).
